packages/extension/src/utils/checkIBCCompliance.ts

```diff
@@ -16,16 +16,25 @@
 
 /**
  * Check if cookie is compliant with IBC(Incrementally Better Cookies) proposal.
- * @param samesite SameSite attribute value.
- * @param secure Secure attribute value.
- * @param chromeCookieStoreHasCookie True if cookie is stored in Chrome store.
+ * @param samesite Cookie SameSite attribute
+ * @param secure Cookie secure attribute
+ * @param name Cookie name
+ * @param url Cookie url
  * @returns {boolean} True if cookie is compliant with IBC proposal.
  */
-export function checkIBCCompliance(
+export async function checkIbcCompliance(
   samesite = 'lax',
   secure = false,
-  chromeCookieStoreHasCookie = false
-): boolean {
+  name: string,
+  url: string
+): Promise<boolean> {
+  const chromeCookieStoreHasCookie = Boolean(
```

Comment:
What are those cases when a cookie is not set in the Cookie Store? Is it only about `SameSite` or something else too? Please add those details to the PR description.